src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -983,92 +985,145 @@ private static DocumentTombstone TableValueToTombstone(JsonOperationContext cont
             }
         }
 
-        public struct DeleteOperationResult
-        {
-            public long Etag;
-            public CollectionName Collection;
-        }
-
         public DeleteOperationResult? Delete(DocumentsOperationContext context,
             Slice loweredKey,
             string key,
             long? expectedEtag,
             long? lastModifiedTicks = null,
-            ChangeVectorEntry[] changeVector = null)
+            ChangeVectorEntry[] changeVector = null,
+            LazyStringValue collection = null)
         {
-            // TODO: Should we call TableValueToDocument from here? Or it should be eascape the blittable building as of the data as we don't use it here?
-            var result = GetDocumentOrTombstone(context, loweredKey, throwOnConflict: false);
-            if (result.Item2 != null)
-                return null; //NOP, already deleted
+            var local = GetDocumentOrTombstone(context, loweredKey, throwOnConflict: false);
+            var currentLastTicks = DateTime.UtcNow.Ticks;
+            long etag = -1;
+            var collectionName = collection != null ? new CollectionName(collection) : null;
 
-            var doc = result.Item1;
-            long etag;
-            CollectionName collectionName;
-            if (doc == null)
+            if (_conflictCount > 0)
             {
-                if (_conflictCount > 0) //no tombstone and no document. perhaps we have a conflict?
+                if (local.Item2 != null && local.Item1 != null)
+                {
+                    // Something is wrong, we can't have conflicts and local document/tombstone
```

Comment:
Isn't this actually a document & tombstone at the same time, not tomstone & conflict?